React/src/components/EditingPopup.tsx

```diff
@@ -0,0 +1,83 @@
+import DataGrid, { Editing, Paging, Popup } from "devextreme-react/data-grid";
+import { ToolbarItem } from "devextreme-react/popup";
+import { ClickEvent } from "devextreme/ui/button";
+import { useCallback, useRef } from "react";
+import service from "../data";
+
+import notify from "devextreme/ui/notify";
+
+const employees = service.getData();
+
+function EditingPopup(){
+    const grid = useRef(null);
+    const saveButtonClick = useCallback((e:ClickEvent) => {
+        (grid.current as any).instance.saveEditData();
+    },[])
+    const saveOptions = {
```

Comment:
You need either to memorize consts or move them outside the component where it's possible